public/app/plugins/datasource/grafana-pyroscope-datasource/datasource.ts

```diff
@@ -48,8 +48,15 @@ export class PyroscopeDataSource extends DataSourceWithBackend<Query, PyroscopeD
     });
   }
 
-  async getProfileTypes(): Promise<ProfileTypeMessage[]> {
-    return await this.getResource('profileTypes');
+  async getProfileTypes(start: number, end: number): Promise<ProfileTypeMessage[]> {
+    return await this.getResource('profileTypes', {
+      start,
+      end,
+    });
+  }
+
+  async getAllProfileTypes(): Promise<ProfileTypeMessage[]> {
+    return await this.getProfileTypes(0, 0);
```

Comment:
I would say it's much cleaner to not provide the start/end time if we don't want to use them and we want to show all profiles. Is there a reason we went with this approach or is it more a case of reflecting how the API works?

Reply:
Fair enough 👍 in https://github.com/grafana/grafana/pull/77523/commits/43da2ba156b452919a91ab9fd0f865dec5cfd533 I changed it to not pass start/end. Both the Grafana backend and the Pyroscope API will translate no start/end into "select everything" requests.